lib/mpool/src/mblock_file.c

```diff
@@ -1255,6 +1255,7 @@ mblock_file_info_get(struct mblock_file *mbfp, struct mblock_file_info *info)
     if (rc == -1)
         return merr(errno);
 
+    info->allocated = 512 * sbuf.st_blocks;
```

Comment:
Where did this 512 come from? Weird how this was missing and no one noticed. Is there not a test that needs changing or something?

Reply:
man fstat
blkcnt_t  st_blocks;      /* Number of 512B blocks allocated */

I think this was accidentally removed a week back in the introspection PR...

Reply:
Very strange. Thanks for context.

Reply:
Our current unit tests check only for the allocated metadata bytes. Let me add a test case to validate the allocated data bytes as well.

Reply:
What is the difference between st.st_blksize and 512? What is the significance of 512? Is it always 512?

Reply:
I think 512 is historical as it's the sector size which the Linux block layer works with.. So, I guess it's not portable...

Reply:
Extended a unit test to check for allocated data bytes.

Reply:
unsigned long  st_blksize;

Reply:
wait, shouldn't you use st_size?  if it's a sparse file then st_blocks will be small, possibly zero..

Reply:
Right, st_blksize is FS specific.

Good call on S_BLKSIZE, seems to be there just for this purpose -
# define S_BLKSIZE      512     /* Block size for `st_blocks'.  */

S_BLKSIZE is is not used by the kernel and is platform dependent. But I think we can use it safely in our code?

Reply:
Sounds like the right decision for us.

Reply:
S_BLKSIZE also exist in FreeBSD (and also is 512), seems right...

Reply:
Done.